src/main/java/hudson/plugins/copyartifact/Copier.java

```diff
@@ -34,8 +39,34 @@
      * @param srcDir Source for upcoming file copy
      * @param baseTargetDir Base target dir for upcoming file copy (the copy-artifact
      *   build step may later specify a deeper target dir)
+     *
+     * @since TODO ?
      */
-    public abstract void init(Run src, AbstractBuild<?,?> dst, FilePath srcDir, FilePath baseTargetDir) throws IOException, InterruptedException;
+    public void init(Run src, Run<?,?> dst, FilePath srcDir, FilePath baseTargetDir) throws IOException, InterruptedException {
```

Comment:
Was part of original code and I was avoiding making unrelated changes as requested. Will fix of course.